sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/InsertIntoHadoopFsRelationCommand.scala

```diff
@@ -263,4 +280,107 @@ case class InsertIntoHadoopFsRelationCommand(
       }
     }.toMap
   }
+
+  /**
+   * Detect the conflict when there are several InsertIntoHadoopFsRelation operations
+   * write to same partition in the same table or a non-partitioned table concurrently.
+   */
+  private def detectConflict(
+      fs: FileSystem,
+      path: Path,
+      staticPartitionKVs: Seq[(String, String)]): Unit = {
+    for (i <- 0 until partitionColumns.size) {
+      Some(".spark-staging-" + i)
```

Comment:
All this is basically `val stagingDir = new Path(...)`.

Reply:
thanks